distributed/tests/test_utils_comm.py

```diff
@@ -57,7 +58,7 @@ async def test_gather_from_workers_permissive_flaky(c, s, a, b):
     assert bad_workers == [a.address]
 
 
-def test_retry_no_exception(loop):
+def test_retry_no_exception(cleanup):
```

Comment:
I see the `cleanup` fixture also yields a pristine loop for sync test functions. It seems `gen_test`  is the counterpart for async functions?

A nice to have might be to have `gen_test_async` and `gen_test_sync` decorators, but I realise I don't have a lot of context here. This shouldn't block the PR.

Reply:
I'll be removing the call to pristine_loop in cleanup in my next PR